articles/142_idl.md

```diff
@@ -35,3 +35,53 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Type Mapping
+
+The next table defines how IDL types are mapped to the following programming languages:
+
+* C11 (or higher)
+* C++11 (or higher)
+* Python 3 (or higher)
+
+| IDL type    | C type      | C++ type    | Python type          |
+| --------    | ----------- | ----------- | -------------------- |
+| float       | float       | float       | float                |
+| double      | double      | double      | float                |
+| long double | long double | long double | float                |
+| char        | signed char | char        | str with length 1    |
+| wchar       | char16\_t   | char16\_t   | str with length 1    |
+| bool        | \_Bool      | bool        | bool                 |
+| octet       | uint8\_t    | uint8\_t    | bytes with length 1  |
```

Comment:
Looking at newer specs I was wondering if the C++ mapping should be `byte` (available as of C++17) instead. To match the semantic C could then be mapped to unsigned `char`.

[We would need to check for availability across all targeted platforms. Even if not all support the type yet we could still make the decision today (using a custom typedef or using `char` directly) and switch to the standard type once available.]

Reply:
Looking at [cppreference](https://en.cppreference.com/w/cpp/compiler_support) and [Visual Studio conformance announcement](https://blogs.msdn.microsoft.com/vcblog/2018/05/07/announcing-msvc-conforms-to-the-c-standard/) it looks like all compilers support it